test/integration_test/kubevirt_test.go

```diff
@@ -98,14 +98,33 @@ func validateVM(t *testing.T, virtualMachine kubevirtv1.VirtualMachine, vmName s
 }
 
 func createTemplatePVC() error {
-	_, err := schedulerDriver.Schedule("",
+	// Check if template PVCs already exist
+	pvcsPreTemplate, err := core.Instance().GetPersistentVolumeClaims(kubevirtTemplateNamespace, nil)
+	if err != nil {
+		return fmt.Errorf("failed to get PVCs from %s namespace", kubevirtTemplateNamespace)
+	}
+
+	_, err = schedulerDriver.Schedule("",
 		scheduler.ScheduleOptions{
 			AppKeys:   []string{"kubevirt-templates"},
 			Namespace: kubevirtTemplateNamespace,
 		})
 	if err != nil {
 		return fmt.Errorf("error deploying kubevirt templates")
 	}
+
+	pvcsPostTemplate, err := core.Instance().GetPersistentVolumeClaims(kubevirtTemplateNamespace, nil)
+	if err != nil {
+		return fmt.Errorf("failed to get PVCs from %s namespace post templates", kubevirtTemplateNamespace)
+	}
+
+	// If number of PVCs is the same in the template namespace that means no new PVCs are created
+	// therefore no importer pods will be run, hence exit the function
+	if len(pvcsPreTemplate.Items) == len(pvcsPostTemplate.Items) {
+		logrus.Infof("No new PVCs were created for templates. Won't wait for importer pods since they will not be started.")
+		return nil
+	}
+
```

Comment:
We seem to ignore errors from the importer pod. If the state is not "running", we assume that it completed successfully. But what if the importer pod encountered an error?

So, I am wondering if we can just ignore the importer pod and check the pvc annotations. I observed that the PVC annotations change as below:
```
$ diff pvc.before pvc.after
5,7c5,7
<     cdi.kubevirt.io/storage.condition.running: "true"
<     cdi.kubevirt.io/storage.condition.running.message: ""
<     cdi.kubevirt.io/storage.condition.running.reason: Pod is running
---
>     cdi.kubevirt.io/storage.condition.running: "false"
>     cdi.kubevirt.io/storage.condition.running.message: Import Complete
>     cdi.kubevirt.io/storage.condition.running.reason: Completed
10c10
<     cdi.kubevirt.io/storage.pod.phase: Running
---
>     cdi.kubevirt.io/storage.pod.phase: Succeeded
``` 
We can look at the `cdi.kubevirt.io/storage.pod.phase` on the PVC. Also, we already hardcode the namespace in this code. Can we also just get the PVC by name and look at the annotations?